app/models/user.rb

```diff
@@ -518,7 +518,12 @@ def any_kind_of_delegate?
   end
 
   def trainee_delegate?
-    delegate_role_metadata.trainee_delegate.any?
+    # NOTE: it used to be `delegate_role_metadata.trainee_delegate.any?`, but
+    # it does fire a db query even if the `delegate_role_metadata` is eager
+    # loaded.
+    delegate_role_metadata.any? do |r|
+      r.status == RolesMetadataDelegateRegions.statuses[:trainee_delegate]
+    end
```

Comment:
Because the Delegate status column is backed by a Rails `enum`, it gives you `r.trainee_delegate?` as a convenience method for free. So I think this can be turned into `delegate_role_metadata.any?(&:trainee_delegate?)` as a shorthand.

Reply:
With that being said, your approach makes it much more inefficient for cases where the data is _not_ pre-loaded. Your above change uses `includes` by default, so it shouldn't be much of a problem, but I guess I am mostly astonished by Rails that it doesn't seem to "respect" pre-loading for the `exists?` method.